docs/vm-installation.md

```diff
@@ -0,0 +1,114 @@
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file to access `antrea-controller` API server. VM should have network
+   connectivity to reach antrea-controller server
+
+```bash
+export ANTREA_API_SERVER=<Antrea API service endpoint>
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-cluster antrea --server=$ANTREA_API_SERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig set-context antrea-agent@antrea --cluster=antrea --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.antrea.kubeconfig use-context antrea-agent@antrea
+```
+
+## Prerequisites on Linux VM
+
+OVS needs to be installed. Please refer to the [getting-started guide](getting-started.md#open-vswitch)
+
+## Running antrea-agent on Linux VM
+
+1. Generate `antrea-agent` binaries
+
+```bash
+make docker-bin
+```
+
+2. Create `antrea-agent` service, below is a sample snippet to start `antrea-agent` as a service on Ubuntu:
+
+```bash
+mkdir -p /var/log/antrea
+cat << EOF > /etc/systemd/system/antrea-agent.service
+Description="antrea-agent as a systemd service"
+After=network.target
+[Service]
+ExecStart=/usr/sbin/antrea-agent \
+--config=/root/antrea/antrea-agent.conf \
+--logtostderr=false \
+--log_file=/var/log/antrea/antrea-agent.log
+Restart=on-failure
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo systemctl daemon-reload
+sudo systemctl enable antrea-agent
+sudo systemctl start antrea-agent
+```
+
+3. The `antrea-agent` configuration file specifies agent configuration parameters. Please refer to this [base configuration file](../build/yamls/externalnode/conf/antrea-agent.conf)
+   to check all the configuration parameters for VMs. Update `clientConnection` and `antreaClientConnection` with the
+   correct kubeconfig file path. Also update `externalNodeNamespace` value to vm-ns
+
+## Prerequisites on Windows VM
+
+1. OVS needs to be installed. For more information refer to the [Antrea Windows documentation](windows.md#1-optional-install-ovs-provided-by-antrea-or-your-own).
+2. Enable Hyper-V optional feature on Windows VM.
+
+```powershell
+Install-WindowsFeature Hyper-V-Powershell
+Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All -NoRestart
+```
+
+3. Download [nssm](https://nssm.cc/download) to configure antrea-agent as a service.
+
+## Running antrea-agent on Windows VM
+
+1. Generate `antrea-agent` binaries
+
+```powershell
+make docker-windows-bin
```

Comment:
We should support downloading the binaries from release assets directly.

Reply:
Ok. We may add downloading after the release.